### PR TITLE
API-24936-Replace-bgs-ext-claims

### PR DESCRIPTION
**Pull request by stiehlrod**
## Summary

As we have identified the Claims API latency issue to be related to the bgs-ext gem, we will need to move our BGS services for claim status to Faraday as well.

Acceptance Criteria:

Move /findBenefitClaimsStatusByParticipantId to Faraday
Move /findBenefitClaimDetailsByBnftClaimId to Faraday

- team Dash
- *(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)*

## Related issue(s)
- [API-24936](https://vajira.max.gov/browse/API-24936)
- *Link to ticket created in va.gov-team repo*
- *Link to previous change of the code/bug (if applicable)*
- *Link to epic if not included in ticket*

### CODE REVIEW

File: modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb [hunk 1]
```diff
@@ -40,6 +41,18 @@ def show
 
         private
 
+        def bgs_service
+          BGS::Services.new(external_uid: target_veteran.participant_id,
+                            external_key: target_veteran.participant_id)
+        end
```
Comment (mchristiansonVA) [inline modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb:47]: I noticed that bgs_service is also defined in claims_api/v2/application_controller.rb; in v1 claims it's defined in claims_api/v1/forms/base.rb. Wondering if we should standardize where we define external services? Following this pattern I would need to also add local_bgs_service to intent_to_file_controller in order to use it for ITF.

File: modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb [hunk 2]
```diff
@@ -143,9 +156,15 @@ def find_lighthouse_claim!(claim_id:)
         def find_bgs_claim!(claim_id:)
           return if claim_id.blank?
 
-          bgs_service.ebenefits_benefit_claims_status.find_benefit_claim_details_by_benefit_claim_id(
-            benefit_claim_id: claim_id
-          )
+          if Flipper.enabled? :bgs_via_faraday
```
Comment (mchristiansonVA) [inline modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb:159]: Do we still need to check the Flipper feature at this point, or are we committed to using the Faraday approach?

File: modules/claims_api/spec/requests/v2/veterans/claims_request_spec.rb [hunk 3]
```diff
@@ -988,36 +996,36 @@
...
+        #             expect(response.status).to eq(200)
+        #             expect(json_response).to be_an_instance_of(Hash)
+        #             expect(json_response['data']['id']).to eq('600236068')
+        #             # expect(first_doc_id).to eq(325_525)
+        #           end
+        #         end
+        #       end
+        #     end
+        #   end
+        # end
```
Comment (FonzMP) [inline modules/claims_api/spec/requests/v2/veterans/claims_request_spec.rb:1028]: Is this supposed to be commented out?

File: modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb (hunk 2 above)
Comment (FonzMP) [reply modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb:159]: yeah, we are safe to remove the Flipper check. Our initial logic with Flipper was set to introduce this in a safe manner. now that we know the connection is working, lets remove this going forward.

**Review by mchristiansonVA — DISMISSED** [review]
LGTM

File: modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb [hunk 4]
```diff
@@ -144,10 +143,10 @@ def find_lighthouse_claim!(claim_id:)
         def find_bgs_claim!(claim_id:)
           return if claim_id.blank?
 
-          bgs_service.ebenefits_benefit_claims_status.find_benefit_claim_details_by_benefit_claim_id(
-            benefit_claim_id: claim_id
+          local_bgs_service.find_benefit_claim_details_by_benefit_claim_id(
+            claim_id
           )
-        rescue Savon::SOAPFault => e
+        rescue Faraday::HttpError::NotFound => e
```
Comment (FonzMP) [inline modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb:149]: huh, these never get hit. if you provide an invalid Claim ID, it returns an empty object.

```
(byebug) local_bgs_service.find_benefit_claim_details_by_benefit_claim_id(claim_id)
{}
```

**Review by FonzMP — APPROVED** [review]
:shipit:

**Pull request merged**